src/System.Collections.Concurrent/src/System/Collections/Concurrent/ConcurrentDictionary.cs

```diff
@@ -82,32 +76,31 @@ internal Tables(Node[] buckets, object[] locks, int[] countPerLock)
         /// </summary>
         private static bool IsValueWriteAtomic()
         {
-            Type valueType = typeof(TValue);
-
-            // We don't have Type.IsClass, so we'll check if it's a reference type by seeing if it gets boxed;
-            bool isClass = (object)default(TValue) == null;
-
             //
             // Section 12.6.6 of ECMA CLI explains which types can be read and written atomically without
             // the risk of tearing.
             //
             // See http://www.ecma-international.org/publications/files/ECMA-ST/Ecma-335.pdf
             //
+            Type valueType = typeof(TValue);
             bool isAtomic =
-                isClass
-                || valueType == typeof(Boolean)
-                || valueType == typeof(Char)
-                || valueType == typeof(Byte)
-                || valueType == typeof(SByte)
-                || valueType == typeof(Int16)
-                || valueType == typeof(UInt16)
-                || valueType == typeof(Int32)
-                || valueType == typeof(UInt32)
-                || valueType == typeof(Single);
+                !valueType.GetTypeInfo().IsValueType
+                || valueType == typeof(bool)
+                || valueType == typeof(char)
+                || valueType == typeof(byte)
+                || valueType == typeof(sbyte)
+                || valueType == typeof(short)
+                || valueType == typeof(ushort)
+                || valueType == typeof(int)
+                || valueType == typeof(uint)
+                || valueType == typeof(float);
 
             if (!isAtomic && IntPtr.Size == 8)
             {
-                isAtomic |= valueType == typeof(Double) || valueType == typeof(Int64);
+                isAtomic =
+                    valueType == typeof(double) ||
```

Comment:
Nit: should the placement of `||` (line prefix vs. suffix) be consistent with lines 88-96 above (either way)?